dashboard/app/models/course.rb

```diff
@@ -177,10 +177,26 @@ def assignable_info
     info
   end
 
-  # Get the set of valid courses for the dropdown in our sections table. This should
-  # be static data, but contains localized strings so we can only cache on a per
-  # locale basis
-  def self.valid_courses
+  # Get the assignable info for this course, substituting any alternate scripts
+  # based on any experiments this user has enabled, then update translations.
+  # @return AssignableInfo
+  def assignable_info_for_user(user)
+    info = ScriptConstants.assignable_info(self)
+    # ScriptConstants gives us untranslated versions of our course name, and the
+    # category it's in. Set translated strings here
+    info[:name] = localized_title
+    info[:category] = I18n.t('courses_category')
+    info[:script_ids] = scripts_for_user(user).map(&:id)
+    info
+  end
```

Comment:
Seems like we could share this with our other assignable_info method, and just have
`info[:script_ids] = user ? scripts_for_user(user).map(&:id) : default_course_scripts.map(&:script_id)